ogre2/src/Ogre2MeshFactory.cc

```diff
@@ -436,12 +445,20 @@ bool Ogre2MeshFactory::LoadImpl(const MeshDescriptor &_desc)
         }
 
         // Add all texture coordinate sets
-        for (unsigned int k = 0u; k < subMesh.TexCoordSetCount(); ++k)
+        if (subMesh.TexCoordSetCount() == 0u)
+        {
+          *vertices++ = 0;
```

Comment:
Do we need to check if the mesh has an associated texture before assigning (0,0) texture coordinate? Will there ever be a situation where this can cause bad access?

Reply:
this should be fine. Given that we declare that the mesh will now always have texcoords, we have to add these texcoords regardless of whether the mesh has a texture or not.